draft-barnes-acme.md

```diff
@@ -431,30 +370,134 @@ to, as defined in the below table:
 | Revoke certificate   | revoke-cert      |
 | Registration         | reg              |
 | Authorization        | authz            |
-| Certificate          | cert             |
 | Challenge            | challenge        |
+| Certificate          | cert             |
 
 Other fields in ACME request bodies are described below.
 
-The following table illustrates a typical sequence of requests required to
-establish a new account with the server, prove control of an identifier, issue a
-certificate, and fetch an updated certificate some time after issuance.  The
-"->" is a mnemonic for a Location header pointing to a created resource.
-
-| Action             | Request        | Response     |
-|:-------------------|:---------------|:-------------|
-| Register           | POST new-reg   | 201 -> reg   |
-| Request challenges | POST new-authz | 201 -> authz |
-| Answer challenges  | POST challenge | 200          |
-| Poll for status    | GET  authz     | 200          |
-| Request issuance   | POST new-cert  | 201 -> cert  |
-| Check for new cert | GET  cert      | 200          |
-
 ACME servers that are intended to be generally accessible need to use
 Cross-Origin Resource Sharing (CORS) in order to be accessible from
 browser-based clients {{W3C.CR-cors-20130129}}.  Such servers SHOULD set the
 Access-Control-Allow-Origin header field to the value "*".
 
+## Registration Objects
+
+An ACME registration resource represents a set of metadata associated to an
+account key pair.  Registration resources have the following structure:
+
+key (required, dictionary):
+: The public key of the account key pair, encoded as a JSON Web Key object
+{{RFC7517}}.
+
+contact (optional, array of string):
+: An array of URIs that the server can use to contact the client for issues
+related to this authorization. For example, the server may wish to notify the
+client about server-initiated revocation.
+
+agreement (optional, string):
+: A URI referring to a subscriber agreement or terms of service provided by the
+server (see below).  Including this field indicates the client's agreement with
```

Comment:
I think this reference is dangling; I don't see (any more) a section about the ToS except here.